### PR TITLE
Multi package docs

### PR DESCRIPTION
**Pull request by lewtun**
# What does this PR do?

This PR overhauls the current doc build to support docs generated from subpackages like `optimum-habana`. The basic idea is to generate the docs for `optimum` and each subpackage via separate Docker containers and stitch together all the HTML files / `_toctree.yml` sections in a post-processing step.

You can see the resulting docs in the screenshot below (or by clicking on the doc build link below):

Screen Shot 2022-07-17 at 21 05 42

I also took the liberty to update the doc build actions to line up with the current state in `doc-builder`.

## TODO

- [x] Merge https://github.com/huggingface/optimum-habana/pull/68 before merging this PR
- [x] Test doc build on `main` via dummy `doc-builder-*` branch

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -0,0 +1,407 @@
...
+```
+
+The docstring should give a minimal, clear example of how the respective model 
+is to be used in inference and also include the expected (ideally sensible)
+output.
+Often, readers will try out the example before even going through the function 
+or class definitions. Therefore it is of utmost importance that the example 
+works as expected.
+
+# Adding documentation support for an 🤗 Optimum subpackage
```
Comment (lewtun) [inline docs/README.md:312]: Here's some details on how to add docs for `optimum-intel` and `optimum-graphcore`

File: docs/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,8 @@
+FROM nikolaik/python-nodejs:python3.8-nodejs18
```
Comment (lewtun) [inline docs/Dockerfile:1]: We need `node` to build the HTML files of the docs. Not sure if this is the best way to have both Python and Node in the same build

File: .github/workflows/build_main_documentation.yml [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+      - name: Create build directory
+        run: |
+          mkdir -p build_dir/optimum
+          cd build_dir/optimum
+          wget https://raw.githubusercontent.com/huggingface/doc-build/main/optimum/_versions.yml
+
+      - name: Make Habana documentation
+        run: |
+          cd optimum-habana
+          git fetch origin
```
Comment (lewtun) [inline .github/workflows/build_main_documentation.yml:71]: These lines should be removed when https://github.com/huggingface/optimum-habana/pull/68 is merged

File: .github/workflows/build_pr_documentation.yml [hunk 4]
```diff
@@ -28,63 +27,80 @@ jobs:
...
+        run: |
+          mkdir build_dir
+          mkdir build_dir/optimum
+          cd build_dir/optimum
+          wget https://raw.githubusercontent.com/huggingface/doc-build/main/optimum/_versions.yml
+
+      - name: Make Habana documentation
+        run: |
+          cd optimum-habana
+          git fetch origin
```
Comment (lewtun) [inline .github/workflows/build_pr_documentation.yml:58]: These lines should be removed when https://github.com/huggingface/optimum-habana/pull/68 is merged

File: docs/README.md [hunk 5]
```diff
@@ -0,0 +1,422 @@
...
+followed by its type, with its shape if it is a tensor, a colon and its
+description:
+
+```
+    Args:
+        n_layers (`int`): The number of layers of the model.
+```
+
+If the description is too long to fit in one line, another indentation is
+necessary before writing the description after th argument.
```
Comment (regisss) [inline docs/README.md:157]:
```suggestion
necessary before writing the description after the argument.
```

File: docs/README.md [hunk 6]
```diff
@@ -0,0 +1,422 @@
...
+- all code examples are formatted using black, like the code of the 🤗 Optimum
+  library
+
+This script may have some weird failures if you made a syntax mistake or if you
+uncover a bug. Therefore, it's recommended to commit your changes before running
+`make style`, so you can revert the changes done by that script easily.
+
+## Testing documentation examples
+
+Good documentation oftens comes with an example of how a specific function or
```
Comment (regisss) [inline docs/README.md:265]:
```suggestion
Good documentation often comes with an example of how a specific function or
```

**Review by regisss — APPROVED** [review]
Thank you very much @lewtun, that looks great :fire: 
I just noticed a couple of typos and left a nit.

**Review by echarlaix — COMMENTED** [review]
LGTM, thanks for taking care of this @lewtun that's awesome 🌟

**Review by echarlaix — APPROVED** [review]

File: .github/workflows/build_pr_documentation.yml (hunk 4 above)
Comment (regisss) [reply .github/workflows/build_pr_documentation.yml:58]: This PR has been merged!

File: .github/workflows/build_main_documentation.yml (hunk 3 above)
Comment (regisss) [reply .github/workflows/build_main_documentation.yml:71]: This PR has been merged!

**Pull request merged**